### ISSUE TITLE
Can't have different flyway and default migration schema since version 8.5.0

### ISSUE THREAD

**Issue by VTLob** [flyway/flyway#3493]
Scenario: have flyway schema history table in its own schema (e.g. flyway) while the default schema to apply the database migrations against is the database default one (e.g. public).

Up until 8.4.4 this was achieved by having _defaultSchema=flyway_ and _schemas=public,flyway_ cause _schemas[0]_ was used as the default migration schema. Since 8.5.0 this is no longer the case and defaultSchema is used for both.

Naming wise it makes sense but sadly this change removes the option of having flyway schema history table in a schema that is not the default migration schema.

If this change was intended then please feel free to take this issue as a feature request. In case there is another way to achieve this (or if this issue is addressed in 9.x which I can't use yet) please let me know.

**Comment by JasonLuo-Redgate** [comment]
Hi @VTLob , may I know which type of database you are using?

**Comment by VTLob** [comment]
This specific example is using a postgres database with default schema "_public_" and flyway migration in schema "_flyway_".

**Comment by bestis** [comment]
What feedback is this waiting?

I would guess that this is pretty basic use case and like really basic need. 
1. I want to my migrations to be run default to `public` schema.
2. I want the `flyway_schema_history` not to be in `public` schema.

And now this isn't possible? Database is Postgresql.

One can do ugly things, like use `beforeEachMigrate__` to change the schema to public, but this feels so basic use case that I'm really surprised that this can't be done.

**Comment by JasonLuo-Redgate** [comment]
Hi @bestis , sorry I forgot to remove the `waiting for feedback` tag from the ticket. 
Yes, we have recognized this a feature. And currently this only works the way in SQL Server, not in Postgresql.

**Comment by ksrkkr** [comment]
Hello, @JasonLuo-Redgate do you plan to add this feature to Postgresql database?

**Comment by hgschmie** [comment]
I am migrating an existing app from flywaydb 8.0.x to current flywaydb. Has there been any further consideration given to this? 

Basically, adding a config parameter "historySchema" or so that, if set, overrides the defaultSchema for the history table seems to be sufficient.

**Comment by OksiBlack** [comment]
Also interested in the topic, I think such feature will be very appreciated and often used.

**Comment by OksiBlack** [comment]
Any news on that?

**Comment by bendlas** [comment]
I need this feature as well

**Comment by mferretti** [comment]
yeah, me too ... especially because i convinced my money spending reference to give PostgreSQL a shot adding the fact that we had schemas and all the flyway bits and bolts could be separated from the actual data ...

**Comment by Miggets7** [comment]
Not sure if this issue I posted is related?
https://stackoverflow.com/questions/79767372/flywaydb-doesnt-create-tables-in-specified-schema
